### PR TITLE
Debugging docs

### PR DESCRIPTION
**Pull request by olydis**
...and refactored main README.md: It called out some things from the "architecture" folder, but not all, and arguably not the most important ones. Instead, now just references the architecture overview as a whole. Essentially we listed the same documents in 2 places but then failed to keep both up to date.

### CODE REVIEW

File: docs/developer/debugging-autorest.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Debugging a classic generator
+
+The first generation of AutoRest generators were written in C#, resulting in the `AutoRest.dll` which now merely serves as one of many [extensions](./architecture/AutoRest-extension.md) to the AutoRest core.
+This extension is not usable or invokable as a stand-alone since it relies on previous processing steps either done by other extensions or by the AutoRest core.
+
+As a result, debugging the extension also requires it to be launched *through the core* like in a regular invocation of AutoRest (on the bright side, one can stay on the CLI and run exactly the command that exposes the problem to debug).
+One can then attach to the running extension process (`dotnet AutoRest.dll --server`) through VS Code, there is a launch configuration for that.
+The biggest problem is timing, i.e. attaching and having breakpoints in place before execution has passed the critical point.
+
+For this purpose, we introduced the `--debugger` flag which will cause any call to `AutoRest.dll` to *sit and wait* until a debugger is attached.
```
Comment (dsgouda) [inline docs/developer/debugging-autorest.md]: Is this a dotnet feature or a flag that the AutoRest.dll knows about?

File: src/autorest-core/test/blaming.ts [hunk 2]
```diff
@@ -13,30 +13,32 @@ import { parse } from "../lib/ref/jsonpath";
...
-  @test @skip @timeout(0) async "end to end blaming with literate swagger"() {
+  @test @timeout(0) async "end to end blaming with literate swagger"() {
     const autoRest = new AutoRest(new RealFileSystem(), ResolveUri(CreateFolderUri(__dirname), "../../test/resources/literate-example/readme-composite.md"));
 
     autoRest.AddConfiguration({ "use-extension": { "@microsoft.azure/autorest-classic-generators": `${__dirname}/../../../core/AutoRest` } })
     // PumpMessagesToConsole(autoRest);
     const view = await autoRest.view;
     assert.equal(await autoRest.Process().finish, true);
 
+    const keys = Object.keys((view.DataStore as any).store);
```
Comment (dsgouda) [inline src/autorest-core/test/blaming.ts]: Are these related changes? If so why

File: docs/developer/debugging-autorest.md (hunk 1 above)
Comment (olydis) [inline docs/developer/debugging-autorest.md]: AutoRest.dll knows about it

File: src/autorest-core/test/blaming.ts (hunk 2 above)
Comment (olydis) [inline src/autorest-core/test/blaming.ts]: oh no, just don't wanna spam PRs... just taking the chance to reenable some tests that were broken over the past weeks due to protocol and format changes.

File: docs/developer/debugging-autorest.md (hunk 1 above)
Comment (dsgouda) [inline docs/developer/debugging-autorest.md]: I'm assuming its already implemented then?

Comment (olydis) [inline docs/developer/debugging-autorest.md]: yep before lunch 🙂

**Comment by olydis** [comment]
@veronicagg @sarangan12 [FYI](https://github.com/Azure/autorest/blob/2.0/docs/developer/debugging-autorest.md), let me know if you have any questions 🙂

**Comment by olydis** [comment]
@laurentmazuel FYI

**Comment by olydis** [comment]
@mcardosos FYI
